Sources/Core/Other/WindowMonitor.swift

```diff
@@ -0,0 +1,128 @@
+//
+//  WindowMonitor.swift
+//  XestiMonitors-iOS
+//
+//  Created by Martin Mungai on 20/03/2018.
+//
+//  © 2016 J. G. Pusey (see LICENSE.md)
+
+#if os(iOS)
+    import Foundation
+    import UIKit
+
+    ///
+    /// A `WindowMonitor` instance monitors the window for changes to its
+    /// visibility or to its frame.
+    ///
+    
+    public class WindowMonitor: BaseNotificationMonitor {
+        ///
+        /// Encapsulates changes to the visibility of the window and to the frame
+        /// of the window.
+        ///
+        public enum Event {
+            
+            case didBecomeVisible(UIWindow)
+            case didBecomeHidden(UIWindow)
+            case didBecomeKey(UIWindow)
+            case didResignKey(UIWindow)
+        }
+        
+        public struct Options: OptionSet {
+            
+            ///
+            /// Monitor `didBecomeVisible` events
+            ///
+            public static let didBecomeVisible = Options(rawValue: 1 << 0)
```

Comment:
Please list properties (except of `all`) in alphabetical order and then assign raw values sequentially.